lib/workers/repository/update/branch/index.ts

```diff
@@ -73,17 +76,38 @@ async function deleteBranchSilently(branchName: string): Promise<void> {
 }
 
 export interface ProcessBranchResult {
+  configAndManagersHash?: string;
   branchExists: boolean;
   prBlockedBy?: PrBlockedBy;
   prNo?: number;
   result: BranchResult;
 }
 
+function canSkipBranchUpdateCheck(
+  branchName: string,
+  branchCache: BranchCache,
+  configAndManagersHash: string
+): boolean {
+  const branchCommit = getBranchCommit(branchName);
+
+  return (
+    branchCommit === branchCache.sha &&
+    configAndManagersHash === branchCache.configAndManagersHash
+  );
+}
+
 export async function processBranch(
-  branchConfig: BranchConfig
+  branchConfig: BranchConfig,
+  branchCache: BranchCache
 ): Promise<ProcessBranchResult> {
   let config: BranchConfig = { ...branchConfig };
   logger.trace({ config }, 'processBranch()');
+  const managersHash = hasha(
+    config.upgrades
+      .map((upgrade) => hashMap.get(upgrade.manager) ?? upgrade.manager)
+      .filter(is.string)
+  );
+  const configAndManagersHash = hasha([JSON.stringify(config), managersHash]);
```

Comment:
makes it sense to move this one layer up too? so we don't have to return the hash? 🤔

Reply:
Yep sounds good. Thanks